### ISSUE TITLE
rescaling eigenvectors at the end of MVMDS computation?

### ISSUE THREAD

**Issue by alexandremerasli** [neurodata/mvlearn#206]
## Expected Behavior
As MVMDS is an extension of cMDS, I expected at the end to rescale the common principal components given what the cMDS rescales the computed eigenvectors by the squareroot of their corresponding eigenvalues.

## Actual Behavior
For now, no such rescaling is done after computing the common principal components in the MVMDS.

## Your Environment
* Python version: 3.7.3
* mvlearn version: 0.2.1

**Comment by alexandremerasli** [comment]
I checked the mvlearn/embed.mvmds.py code and ran it, which gives exactly the same results as the equivalent implementation in the 'multiview' package (cf https://github.com/mariceli3/multiview/blob/master/multiview/mvmds.py).
However, this function is able to compute MVMDS using only 1 view as input, which normally correspond to classical MultiDimensional Scaling-cMDS (for mathematical theory, cf page 9-10 in Groenen, P. J., & Borg, I. (2014). Past, present, and future of multidimensional scaling. Visualization and verbalization of data, 95-117.). When comparing the results with the 'scikit-bio' implementation of the cMDS (cf https://github.com/biocore/scikit-bio/blob/0.5.6/skbio/stats/ordination/_principal_coordinate_analysis.py#L23), the computed maps were different. I went deeper into the code, and saw the difference is the 'mutiview' package does not rescale the common principal components by the squareroot of their corresponding eigenvalues, contrary to the cMDS in 'scikit-bio' (which is the case in the paper cited above).
I would like to understand why both MVMDS implementations do not have this rescaling step.

**Comment by rflperry** [comment]
Yes, cMDS identifies a decomposition of the distance matrix `D = U S U^T` and by viewing `D = XX^T`, then `U` gives us the new space and `S` (eigenvalues) provides the appropriate scaling in that space. Thus, these `X` ideally capture well the distances between samples in a smaller dimension.

 For MVMDS, however, each view has a distance matrix `D_i` and we find the decompositions `D_i = U S_i U^T` which is view specific. The important information across views is `U`. You are correct, however, that this will not yield the cMDS result for a single view as only `U` is returned and not scaled. Each view has its own set of eigenvalues and so one cannot simply scale as is done in cMDS. One could scale by the mean, median, etc. but as far as I am aware this would be situation specific.

Scaling would weigh the components differently which could have significance, say in clustering, and so I concur that such scalings should be added as an option.

**Comment by alexandremerasli** [comment]
That is true that as you have a set of eigenvalues corresponding to each view, you would have to somehow average the eigenvalues if you want to rescale the embedding `U`.

I am currently working on clustering data after applying MVMDS, and as you said, rescaling by the eigenvalues stretches some dimensions, and it does affect clustering. I know the ground truth to determine which clustering was better, and so which MVMDS map to use, but obviously, my objective is to know which MVMDS map to use, with or without rescaling. Do you have any suggestions for choosing? Maybe looking at clustering evaluation indices like the silhouette coefficient and choose the the MVMDS map which gives the higher silhouette coefficient (a bit like choosing K in K-Means)

**Comment by rflperry** [comment]
Silhouette coefficient pertains to unsupervised scenarios, but you indicated you have ground truth so you should be able exactly quantify how good a given clustering is, i.e. [homogeneity score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.homogeneity_score.html).

You could calibrate your clustering algorithm for both without scaling and for some form of scaling (average or eigenvalues of most useful view, etc.) and compare the performance.

Depending on the task, you could also evaluate it as a supervised learning problem and run a classifier on the MVMDS results in a cross-validation setting and see which MVMDS features give the best classification accuracy.

**Comment by alexandremerasli** [comment]
I will try your rescaling suggestions, they seem sensible to me.

Indeed, I have the ground truth but the objective of this study is to see if unsupervised learning can work, and to what extent compared to the supervised approach.

**Comment by alexandremerasli** [comment]
Thanks a lot for your time and ideas! :)
